pay-api/src/pay_api/services/eft_gl_transfer.py

```diff
@@ -0,0 +1,108 @@
+
+from sqlalchemy import func
+
+from pay_api.models import EFTGLTransfer as EFTGLTransferModel
+from pay_api.models import db
+
+
+@dataclass
+class EFTGlTransferSearch:  # pylint: disable=too-many-instance-attributes
+    """Used for searching EFT GL Transfer records."""
+
+    created_on: Optional[datetime] = None
+    invoice_id: Optional[int] = None
+    is_processed: Optional[bool] = None
+    processed_on: Optional[datetime] = None
+    short_name_id: Optional[int] = None
+    source_gl: Optional[str] = None
+    target_gl: Optional[str] = None
+    transfer_type: Optional[str] = None
+    transfer_date: Optional[datetime] = None
+
+
+class EFTGlTransfer:
+    """Service to manage EFT GL transfers."""
+
+    @staticmethod
+    def find_by_id(transfer_id: int) -> EFTGLTransferModel:
+        """Return EFT Transfers by id."""
+        return EFTGLTransferModel.find_by_id(transfer_id)
+
+    @staticmethod
+    def find_by_short_name_id(short_name_id: int, is_processed: bool = None) -> [EFTGLTransferModel]:
+        """Return EFT Transfers by short_name_id."""
+        query = db.session.query(EFTGLTransferModel) \
+            .filter(EFTGLTransferModel.short_name_id == short_name_id)
+
+        if is_processed is not None:
+            query = query.filter(EFTGLTransferModel.is_processed == is_processed) \
+
+        query = query.order_by(EFTGLTransferModel.created_on.asc())
+
+        return query.all()
+
+    @staticmethod
+    def find_by_invoice_id(invoice_id: int, is_processed: bool = None) -> [EFTGLTransferModel]:
+        """Return EFT Transfers by invoice_id."""
+        query = db.session.query(EFTGLTransferModel) \
+            .filter(EFTGLTransferModel.invoice_id == invoice_id)
+
+        if is_processed is not None:
+            query = query.filter(EFTGLTransferModel.is_processed == is_processed)
+
+        query = query.order_by(EFTGLTransferModel.created_on.asc())
+
+        return query.all()
+
+    @staticmethod
+    def search(search_criteria: EFTGlTransferSearch = EFTGlTransferSearch()) -> [EFTGLTransferModel]:
+        """Return EFT Transfers by search criteria."""
+        query = db.session.query(EFTGLTransferModel)
+
+        if search_criteria.created_on:
+            query = query.filter(func.DATE(EFTGLTransferModel.created_on) == search_criteria.created_on.date())
+
+        if search_criteria.invoice_id:
+            query = query.filter(EFTGLTransferModel.invoice_id == search_criteria.invoice_id)
+
+        if search_criteria.is_processed is not None:
+            query = query.filter(EFTGLTransferModel.is_processed == search_criteria.is_processed)
+
+        if search_criteria.processed_on:
+            query = query.filter(func.DATE(EFTGLTransferModel.processed_on) == search_criteria.processed_on.date())
+
+        if search_criteria.short_name_id:
+            query = query.filter(EFTGLTransferModel.short_name_id == search_criteria.short_name_id)
+
+        if search_criteria.source_gl:
```

Comment:
Might be a good idea to write a helper function for this (could also check for date type and handle it for dates)

eg:

`query.filterIfPresent(search_criteria.source_gl, EFTGLTransferModel.short_name_id == search_criteria.short_name_id)`

Reply:
We could use this pattern in auth-api as well, where we have a bunch of conditions like below

Reply:
Updated with CustomQuery to extend the BaseQuery with filter_conditionally helper